bot/exts/moderation/stream.py

```diff
@@ -0,0 +1,135 @@
+    ) -> None:
+        """
+        Stream handles <prefix>stream command.
+
+        argument user - required user mention, any errors should be handled by upper level handler
+        duration - int must be higher than 0 - defaults to 1
+        time_format - str defining what time unit you want to use, must be any of FORMATS - defaults to h
+        Command give user permission to stream and takes it away after provided duration
+        """
+        # Check for required user argument
+        # if not provided send NO_USER_SPECIFIED message
+        if not user:
+            await ctx.send(NO_USER_SPECIFIED)
+            return
+
+        # Time can't be negative lol
+        if duration <= 0:
+            await ctx.send(TIME_LESS_EQ_0)
+            return
+
+        # Check if time_format argument is a valid time format
+        # eg. d, day etc are aliases for day time format
+        if time_format not in FORMATS:
+            await ctx.send(TIME_FORMAT_NOT_VALID)
+            return
+
+        # Check if user already has streaming permission
+        already_allowed = any(Roles.video == role.id for role in user.roles)
+        if already_allowed:
+            await ctx.send(USER_ALREADY_ALLOWED_TO_STREAM)
+            return
+
+        # Set user id - time in redis cache and add streaming permission role
+        await self.user_cache.set(user.id, time.time() + self._parse_time_to_seconds(duration, time_format))
+        await user.add_roles(discord.Object(Roles.video), reason="Temporary streaming access granted")
+        await ctx.send(f"{user.mention} can now stream for {duration} {self._link_from_alias(time_format)[1]}/s")
+
+    @tasks.loop(seconds=30)
+    async def remove_permissions(self) -> None:
+        """Background loop for removing streaming permission."""
+        all_entries = await self.user_cache.items()
+        for user_id, delete_time in all_entries:
+            if time.time() > delete_time:
+                member = self.guild_static.fetch_memebr(user_id)
+                if member:
+                    await member.remove_roles(discord.Object(Roles.video), reason="Temporary streaming access revoked")
+                    await self.user_cache.pop(user_id)
+
+    @remove_permissions.before_loop
+    async def await_ready(self) -> None:
+        """Wait for bot to be ready before starting remove_permissions loop and get guild by id."""
+        await self.bot.wait_until_ready()
+        self.guild_static = self.bot.get_guild(Guild.id)
+
+    @commands.command(aliases=("unstream", ))
+    @commands.has_any_role(*STAFF_ROLES)
+    async def revokestream(
+            self,
+            ctx: commands.Context,
+            user: discord.Member = None
+    ) -> None:
+        """
+        Revokestream handles <prefix>revokestream command.
+
+        argument user - required user mention, any errors should be handled by upper level handler
+        command removes streaming permission from a user
+        """
```

Comment:
The description of the function should always be in the summary section i.e. the first line. Implementation details like error handling need not be documented for something user-facing like a command. The documentation of the `user` argument could be integrated into the summary as it is fairly self-explanatory.

```suggestion
        """Revoke streaming permissions from a user."""
```